packages/libs/client/src/signing/chainweaver/signWithChainweaver.ts

```diff
@@ -7,19 +7,42 @@ import type {
   IQuicksignResponse,
   IQuicksignSigner,
 } from '../../signing-api/v1/quicksign';
-import type { ISignFunction } from '../ISignFunction';
+import type { ISingleSignFunction } from '../ISignFunction';
 import { addSignatures } from '../utils/addSignatures';
 import { parseTransactionCommand } from '../utils/parseTransactionCommand';
 
 const debug: Debugger = _debug('pactjs:signWithChainweaver');
 
+/**
+ * Interface to use when writing a signing function for Chainweaver that accepts a single transaction
+ * @public
+ */
+export interface IChainweaverSingleSignFunction extends ISingleSignFunction {
+  (
+    transaction: IUnsignedCommand,
+    chainweaverUrl?: string,
+  ): Promise<ICommand | IUnsignedCommand>;
+}
+
+/**
+ * Interface to use when writing a signing function for Chainweaver that accepts multiple transactions
+ * @public
+ */
+export interface IChainweaverSignFunction
+  extends IChainweaverSingleSignFunction {
+  (
+    transactionList: IUnsignedCommand[],
+  ): Promise<(ICommand | IUnsignedCommand)[]>;
+}
+
 /**
  * Sign with chainweaver according to {@link https://github.com/kadena-io/KIPs/blob/master/kip-0015.md | sign-v1 API}
  *
  * @public
  */
-export const signWithChainweaver: ISignFunction = (async (
+export const signWithChainweaver: IChainweaverSignFunction = (async (
   transactionList: IUnsignedCommand | Array<IUnsignedCommand | ICommand>,
+  chainweaverUrl = 'http://127.0.0.1:9467',
```

Comment:
This should be done with a wrapper function that creates the `signWithChainweaver` function. The final `signWithX` function should always comply with the `ISignFunction`.

My suggestion is to change this and add the `createSignWithChainweaver` function that'll provide the `chainweaverUrl`.

Reply:
Fixed! This is however now a breaking change 😞

Reply:
We could still expose `signWithChainweaver` and `createSignWithChainweaver` where they both call a non-exposed function that has `chainweaverUrl` as argument where the `signWithChainweaver` uses the native one.

To prevent this from being an issue in the future, we could have a pattern that ALWAYS has the `createSignWithX`-function. Ie: `createSignWithX(options) -> ISignFunction`.

Reply:
`signWithChainweaver` was the only function that didn't follow the `createSignWithX` pattern yet. I'm in favour of the option to always follow this pattern for the sign functions. It's also way more flexible as options can be added as needed without API changes.